### PR TITLE
Expose new Analytics All Traffic widget in production.

### PR DESCRIPTION
**Pull request by benbowler**
## Summary

Addresses issue #2429

## Relevant technical choices

The IB wasn't too clear with the Stories updates, instead of "Remove[ing] Analytics Module/Components/Page Dashboard/All Traffic Widget" I updated it to render with the new component and updated the test data also, inlining it as Felix has in other widget stories. Otherwise, I followed the IB end to end.

## Checklist

- [x] My code is tested and passes existing unit tests.
- [x] My code has an appropriate set of unit tests which all pass.
- [x] My code is backward-compatible with WordPress 4.7 and PHP 5.6.
- [x] My code follows the [WordPress](https://make.wordpress.org/core/handbook/best-practices/coding-standards/) coding standards.
- [x] My code has proper inline documentation.
- [x] I have added a QA Brief on the issue linked above.
- [x] I have signed the Contributor License Agreement (see <https://cla.developers.google.com/>).

### CODE REVIEW

**Comment by benbowler** [comment]
While still blocked by #2422 this is ready for initial code review. Once #2422 is merged this can be passed for QA/MR.

File: assets/js/modules/analytics/components/dashboard/DashboardAllTrafficWidget/UserDimensionsPieChart.js
Comment (felixarntz) [inline assets/js/modules/analytics/components/dashboard/DashboardAllTrafficWidget/UserDimensionsPieChart.js:205]: I noticed a problem while testing, related to this component (see #2425): When I run a development build, the piechart does not show up, in neither the new nor the old widget. I assume that is happening because the chart uses a specific ID but is now used in multiple places?

@eugene-manuilov Do you have an idea how we could fix this? I wonder whether we could somehow increase a static count to prevent duplicate IDs.

Alternatively, if that doesn't work, we could also go with a more pragmatic approach specific to the situation, e.g. `DashboardAllTrafficWidget` could have an optional `isLegacy` prop and based on that the ID used for the inner component's chart would be different - not a great approach, but this situation is temporary anyway, so might be okay. If we can solve it in `UserDimensionsPieChart` internally though, that'd be perfect.

**Review by felixarntz — CHANGES_REQUESTED** [review]
Awesome work! There's only one thing that needs to be addressed here, and two minor comments. It also looks like this broke the e2e tests, so it would be great if you could check those. Potentially it's just that we were relying on some DOM element from the old All Traffic widget somewhere.

File: assets/js/modules/analytics/index.legacy.js
Comment (benbowler) [reply assets/js/modules/analytics/index.legacy.js:88]: Yes, I'm not sure, that was right from the IB. I've reverted it.

File: assets/js/modules/analytics/components/dashboard/DashboardAllTrafficWidget/LegacyDashboardAllTraffic.js
Comment (benbowler) [reply assets/js/modules/analytics/components/dashboard/DashboardAllTrafficWidget/LegacyDashboardAllTraffic.js]: Fixed.

**Comment by benbowler** [comment]
I've fixed the small comments and I'll take a look at the E2E tests now.

File: assets/js/modules/analytics/components/dashboard/DashboardAllTrafficWidget/UserDimensionsPieChart.js
Comment (eugene-manuilov) [reply assets/js/modules/analytics/components/dashboard/DashboardAllTrafficWidget/UserDimensionsPieChart.js:205]: > Do you have an idea how we could fix this? I wonder whether we could somehow increase a static count to prevent duplicate IDs.

@felixarntz I think we can leverage the [`useInstanceId`](https://github.com/WordPress/gutenberg/tree/master/packages/compose#useInstanceId) hook to get a unique id for the `UserDimensionsPieChart` component and then add it as a suffix to the `chartID` prop. What do you think?

Comment (felixarntz) [reply assets/js/modules/analytics/components/dashboard/DashboardAllTrafficWidget/UserDimensionsPieChart.js:205]: @eugene-manuilov SGTM! cc @benbowler

Comment (benbowler) [reply assets/js/modules/analytics/components/dashboard/DashboardAllTrafficWidget/UserDimensionsPieChart.js:209]: @felixarntz this has been [implemented here](https://github.com/google/site-kit-wp/pull/2617/commits/3add375d269f207bea6b3da62d9933e5a23b1923).

**Comment by felixarntz** [comment]
@benbowler Merged `develop` back in which now includes the `UserCountGraph` implemented.

**Review by felixarntz — CHANGES_REQUESTED** [review]
@benbowler @eugene-manuilov Not sure what's going on there. I saw you added `useInstanceId`, but when I run the latest code in a development build, I see only one of the two charts showing up at a time.

They still appear to be connected in some way they shouldn't be. For example, when I switched tabs in the new widget, it scrolled down to the legacy widget and showed the graph there.

**Comment by aaemnnosttv** [comment]
@googlebot I consent.

**Comment by aaemnnosttv** [comment]
@felixarntz 

> when I switched tabs in the new widget, it scrolled down to the legacy widget and showed the graph there.

We had the same problem with the PageSpeed widget; it has to do with the default behavior of the `TabBar`. We just need to configure the `Tab` components to not `focusOnActivate`. I've pushed a commit that fixes this.

> when I run the latest code in a development build, I see only one of the two charts showing up at a time.

This seems to be limited to the initial render/draw. When I click a dimension tab, it appears. Taking a closer look at this now.

**Comment by benbowler** [comment]
Looping back here, this is the only comment outstanding to look at I believe:

> @felixarntz  when I run the latest code in a development build, I see only one of the two charts showing up at a time.

> @aaemnnosttv  This seems to be limited to the initial render/draw. When I click a dimension tab, it appears. Taking a closer look at this now.

**Comment by aaemnnosttv** [comment]
The problem I was seeing before is that the `onReady` callback for the chart wasn't being called for both. Not sure if it consistently works when there's only one either, but I can look into this a bit more.

**Comment by eugene-manuilov** [comment]
@benbowler i think the issue is caused by the `fontName` options. Try to comment out the following lines, those charts should start working correctly:

https://github.com/google/site-kit-wp/blob/df38a87cef996a23f35434400d5eac632b604df8/assets/js/modules/analytics/components/dashboard/DashboardAllTrafficWidgetV2/UserDimensionsPieChart.js#L245

https://github.com/google/site-kit-wp/blob/df38a87cef996a23f35434400d5eac632b604df8/assets/js/modules/analytics/components/dashboard/DashboardAllTrafficWidgetV2/UserDimensionsPieChart.js#L259

**Comment by benbowler** [comment]
@eugene-manuilov's fix has worked for me when testing. I've committed it, removed the now unused `useInstanceID` hook and pushed ready for review.

File: assets/js/modules/analytics/components/dashboard/DashboardAllTrafficWidget/UserDimensionsPieChart.js
Comment (eugene-manuilov) [inline assets/js/modules/analytics/components/dashboard/DashboardAllTrafficWidget/UserDimensionsPieChart.js:212]: @benbowler the `chartID` is important for this chart and it should be unique. Please, use the `useInstanceId` hook again to make the `chartID` property unique on per instance basis. Plus, make sure that `GoogleChart.charts.get( 'user-dimensions-pie-chart' )` calls in the `onReady` callback use the same chart id as we pass to the `GoogleChart` component.

**Review by eugene-manuilov — CHANGES_REQUESTED** [review]

File: assets/js/modules/analytics/components/dashboard/DashboardAllTrafficWidget/UserDimensionsPieChart.js
Comment (benbowler) [reply assets/js/modules/analytics/components/dashboard/DashboardAllTrafficWidget/UserDimensionsPieChart.js:212]: Resolved.

File: assets/js/modules/analytics/components/dashboard/DashboardAllTrafficWidget/LegacyDashboardAllTraffic.js
Comment (eugene-manuilov) [inline assets/js/modules/analytics/components/dashboard/DashboardAllTrafficWidget/LegacyDashboardAllTraffic.js:4]:
```suggestion
 * Site Kit by Google, Copyright 2021 Google LLC
```

File: assets/js/modules/analytics/components/dashboard/DashboardAllTrafficWidget/LegacyDashboardAllTraffic.js
Comment (eugene-manuilov) [inline assets/js/modules/analytics/components/dashboard/DashboardAllTrafficWidget/LegacyDashboardAllTraffic.js:38]:
```suggestion
			<Cell id="googlesitekit-pagespeed-header" size={ 12 }>
```

File: assets/js/modules/analytics/components/dashboard/DashboardAllTrafficWidget/LegacyDashboardAllTraffic.js
Comment (eugene-manuilov) [inline assets/js/modules/analytics/components/dashboard/DashboardAllTrafficWidget/LegacyDashboardAllTraffic.js:47]:
```suggestion
			<Cell size={ 12 }>
```

**Review by eugene-manuilov — CHANGES_REQUESTED** [review]
Thanks, @benbowler! Found a few minor things to change and I think it should be good to go. Could you please take a look at it?

**Comment by benbowler** [comment]
Updated and back to you @eugene-manuilov

**Review by eugene-manuilov — DISMISSED** [review]
Awesome!

**Review by eugene-manuilov — APPROVED** [review]
Looks good now!

**Comment by felixarntz** [comment]
@benbowler @eugene-manuilov FYI I pushed one extra commit here just for some wording changes that came up during review with @marrrmarrr.

**Review by felixarntz — APPROVED** [review]
Awesome work - let's 🚢 !

**Pull request merged**
